components/directory-server-manager/org.wso2.carbon.directory.server.manager.ui/src/main/resources/web/servicestore/delete-finish.jsp

```diff
@@ -79,14 +80,14 @@
         if (!isError) {
     %>
     location.href =
-            '../application/configure-service-provider.jsp?action=delete&display=kerberos&spName=<%=spName%>&kerberos&=<%=servicePrincipleName%>';
+            '../application/configure-service-provider.jsp?action=delete&display=kerberos&spName=<%=Encode.forJavaScriptBlock(Encode.forUriComponent(spName))%>&kerberos&=<%=Encode.forJavaScriptBlock(Encode.forUriComponent(servicePrincipleName))%>';
     <%  } else { %>
-    location.href = '../application/configure-service-provider.jsp?action=cancel&display=kerberos&spName=<%=spName%>';
+    location.href = '../application/configure-service-provider.jsp?action=cancel&display=kerberos&spName=<%=Encode.forJavaScriptBlock(Encode.forUriComponent(spName))%>';
     <%
         }
     }else {
     %>
-    location.href = '<%=forwardTo%>';
+    location.href = '<%=Encode.forUriComponent(forwardTo)%>';
```

Comment:
forwardTo is hardcoded and trusted. It's not necessary to encode here. Further encoding the complete URI may break the URI. We should encode only the query string of a URI. If applying encoding here we should use only javascript block encode